src/Microsoft.Diagnostics.Tools.RuntimeClient/Eventing/EventPipeClient.cs

```diff
@@ -0,0 +1,171 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.IO.Pipes;
+using System.Linq;
+using System.Net.Sockets;
+using System.Runtime.InteropServices;
+using System.Security.Principal;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Diagnostics.Tools.RuntimeClient.Eventing
+{
+    public static class EventPipeClient
+    {
+        // TODO: Make async.
+        // TODO: Required an interface (e.g. "ISerializableToByteArray").
```

Comment:
I'd leave it as byte[] unless there is a clear reason to change it. I'd expect this to end as an internal API rather than public which gives us plenty of flexibility to adjust in the future if needed.